### PR TITLE
Fix `Context has expired` while executing query

### PR DESCRIPTION
**Pull request by yariks5s**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Removed the use of the context during function execution, as it was causing errors when the context was expired

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit c37fc005d39f880b6ab12b01e80954f162817790 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68534/c37fc005d39f880b6ab12b01e80954f162817790/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68497/9dee9ecfb4adee4cff099f13afe61bdc2c38170e/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Functions/FunctionSQLJSON.h [hunk 1]
```diff
@@ -204,13 +204,25 @@ class FunctionSQLJSONHelpers
...
+        function_json_value_return_type_allow_complex = context_->getSettingsRef().function_json_value_return_type_allow_complex;
+        function_json_value_return_type_allow_nullable = context_->getSettingsRef().function_json_value_return_type_allow_nullable;
+    }
 
     static constexpr auto name = Name::name;
+    bool max_parser_depth;
+    bool max_parser_backtracks;
+    bool allow_simdjson;
+    bool function_json_value_return_type_allow_complex;
+    bool function_json_value_return_type_allow_nullable;
```
Comment (kssenii) [inline src/Functions/FunctionSQLJSON.h:225]:
```suggestion
    const bool max_parser_depth;
    const bool max_parser_backtracks;
    const bool allow_simdjson;
    const bool function_json_value_return_type_allow_complex;
    const bool function_json_value_return_type_allow_nullable;
```

File: src/Functions/FunctionSQLJSON.h [hunk 2]
```diff
@@ -204,13 +204,25 @@ class FunctionSQLJSONHelpers
...
+    {
+        max_parser_depth = context_->getSettingsRef().max_parser_depth;
+        max_parser_backtracks = context_->getSettingsRef().max_parser_backtracks;
+        allow_simdjson = context_->getSettingsRef().allow_simdjson;
+        function_json_value_return_type_allow_complex = context_->getSettingsRef().function_json_value_return_type_allow_complex;
+        function_json_value_return_type_allow_nullable = context_->getSettingsRef().function_json_value_return_type_allow_nullable;
+    }
 
     static constexpr auto name = Name::name;
+    bool max_parser_depth;
```
Comment (kssenii) [inline src/Functions/FunctionSQLJSON.h:221]: Also this can be moved to `private`

File: src/Functions/FunctionSQLJSON.h [hunk 3]
```diff
@@ -123,7 +123,7 @@ class FunctionSQLJSONHelpers
     class Executor
     {
     public:
-        static ColumnPtr run(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, uint32_t parse_depth, uint32_t parse_backtracks, const ContextPtr & context)
+        static ColumnPtr run(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count, uint32_t parse_depth, uint32_t parse_backtracks, const bool & function_json_value_return_type_allow_complex)
```
Comment (kssenii) [inline src/Functions/FunctionSQLJSON.h:126]: There is no need in bool reference, please use just bool here and other such cases below

File: src/Functions/FunctionSQLJSON.h (hunk 2 above)
Comment (kssenii) [inline src/Functions/FunctionSQLJSON.h:221]: `max_parser_depth` should be `size_t`, also same for `max_parser_backtracks`

**Review by kssenii — APPROVED** [review]

**Comment by yariks5s** [comment]
- `Upgrade check (debug)` - Not related
- `Performance Comparison (release) [1/4]` - Not related

**Pull request merged**